shamil_latypov/main.cpp

```diff
@@ -79,52 +79,37 @@ class Edge {
   EdgeId id_;
   VertexId v1_;
   VertexId v2_;
-  int color_param_;
+  Color color_;
 };
 
 class Graph {
  public:
   // Возврат значений
   int get_depth() const { return depth_map_.size() - 1; }
-  int get_vertices_cnt_in_depth(int depth) const {
-    return depth_map_[depth].size();
+  const std::vector<VertexId>& get_vertices_in_depth(int depth) const {
+    return depth_map_[depth];
   }
 
   // Добавляет ребро в graph
-  void add_edge(const VertexId& v1, const VertexId& v2) {
-    assert(has_vertex_id(v1) && "Vertex 1 doesnt exist\n");
-    assert(has_vertex_id(v2) && "Vertex 2 doesnt exist\n");
-    assert(!vertices_connected(v1, v2) && "Vertices are connected\n");
-
-    int color_param = 0;
-    // По логике нашей задачи, если вторая вершина не имеет присоединенных к ней
-    // рёбер, значит добавляемое ребро серое
-    if (get_vertex(v2).get_edge_ids().size() == 0) {
-      // vertex2.depth = vertex1.depth + 1
-      get_vertex(v2).set_depth(get_vertex(v1).get_depth() + 1);
-
-      if (depth_map_.size() <= get_vertex(v2).get_depth()) {
-        depth_map_.push_back({});
-      }
-
-      // Добавление на нужную глубину vertex2.id и удаление vertex2.id из
-      // нулевой глубины
-      depth_map_[get_vertex(v2).get_depth()].push_back(v2);
-      for (auto vertex_id_it = depth_map_[0].begin();
-           vertex_id_it != depth_map_[0].end(); vertex_id_it++) {
-        if (*vertex_id_it == v2) {
-          depth_map_[0].erase(vertex_id_it);
-          break;
-        }
-      }
-    } else {  // Иначе задаем цвет ребру
-      color_param = set_edge_color(v1, v2);
+  void add_edge(const VertexId& vertex1_id, const VertexId& vertex2_id) {
+    assert(has_vertex_id(vertex1_id) && "Vertex 1 doesnt exist\n");
+    assert(has_vertex_id(vertex2_id) && "Vertex 2 doesnt exist\n");
+    assert(!vertices_connected(vertex1_id, vertex2_id) &&
+           "Vertices are connected\n");
+
+    auto& vertex1 = get_vertex(vertex1_id);
+    auto& vertex2 = get_vertex(vertex2_id);
+    const auto& color_param = set_edge_color(vertex1, vertex2);
```

Comment:
Странно, я вроде оставлял комментарий, но он куда-то пропал...
`set` - значит присвоить, ты же делаешь наоборот - получаешь данные, это довольно грубая ошибка наименования. Смотри как это работает:
```cpp
struct Vertex {
  set_id(const VertexId& id) {
    id_ = id;
  }

  get_id() const {
    return id_;
  }

 private:
  VertexId id_;
}

const auto id = vertex.get_id();
vertex.set_id(id);
```
Вот так работает логика сеттеров и геттеров.

В твоем случае ты геттер назвал сеттером.
Лучше будет
```suggestion
    const auto& color = get_edge_color(vertex1, vertex2);
```
А еще лучше будет, не "получить", а "вычислить"
```suggestion
    const auto& color = calculate_edge_color(vertex1, vertex2);
```

Reply:
И ещё раз, что у тебя за мания к `_param` суффиксу? Это можно каждую переменную так называть - они все являются параметрами чего-то